demo/full/scripts/controllers/ContentList.jsx

```diff
@@ -32,57 +35,131 @@ const URL_DENOMINATIONS = {
+
+  if (IS_HTTPS) {
+    if (content.url.startsWith("http:")) {
+      name = "[HTTP only] " + name;
+      disabled = true;
+    }
+  } else if (!HAS_EME_APIs &&
+             content.drmInfos &&
+             content.drmInfos.length
+  ) {
+    name = "[HTTPS only] " + name;
+    disabled = true;
+  }
+
+  if (content.live) {
+    name += " (live)";
+  }
+
+  if (content.localContent) {
+    name = "[Local storage] " + name;
+  }
+
+  return { content, name, disabled };
+}
+
 const CONTENTS_PER_TYPE = TRANSPORT_TYPES.reduce((acc, tech) => {
   acc[tech] = contentsDatabase
-    .filter(({ transport }) =>
-      transport === tech.toLowerCase()
-    ).map((content) => {
-      let name = content.name;
-      let disabled = false;
-
-      if (IS_HTTPS) {
-        if (content.url.startsWith("http:")) {
-          name = "[HTTP only] " + name;
-          disabled = true;
-        }
-      } else if (!HAS_EME_APIs && content.drmInfos && content.drmInfos.length) {
-        name = "[HTTPS only] " + name;
-        disabled = true;
-      }
+    .filter(({ transport }) => transport === tech.toLowerCase())
+    .map(formatContentForDisplay);
 
-      if (content.live) {
-        name += " (live)";
-      }
-
-      return { content, name, disabled };
-    });
   return acc;
 }, {});
 
 Object.keys(CONTENTS_PER_TYPE).forEach((key) => {
-  CONTENTS_PER_TYPE[key].push({ name: "Custom link", disabled: false });
+  CONTENTS_PER_TYPE[key].unshift({ name: "Custom link", disabled: false });
 });
 
 class ContentList extends React.Component {
   constructor(...args) {
     super(...args);
 
     const contents = CONTENTS_PER_TYPE[TRANSPORT_TYPES[0]];
-    const firstEnabledContentIndex =
-      contents.findIndex((content) => !content.disabled);
+    const firstEnabledContentIndex = contents.length > 1 ?
+      contents.findIndex((c) => !c.disabled && c.name !== "Custom link") :
+      0;
+
+    const localStorageContents = [];
+
+    if (hasLocalStorage) {
+      const localContentItems = localStorage.getItem("rxPlayerLocalContents");
+      if (hasLocalStorage && localContentItems) {
+        try {
+          localStorageContents.push(...JSON.parse(localContentItems));
+        } catch(err) {
+          /* eslint-disable-next-line */
+          console.warn("Demo: Can't parse local storage content.");
+        }
+      }
+    }
 
     this.state = {
       transportType: TRANSPORT_TYPES[0],
       contentChoiceIndex: firstEnabledContentIndex,
-      hasTextInput: CONTENTS_PER_TYPE[TRANSPORT_TYPES[0]].length - 1 ===
-        firstEnabledContentIndex,
+      hasTextInput: firstEnabledContentIndex === 0,
       displayDRMSettings: false,
-      manifestUrl: "",
+      customManifestUrl: "",
+      customContentName: "",
       drm: DRM_TYPES[0],
       autoPlay: true,
+      localStorageContents,
+      isSavingOrUpdating: false,
+      chosenContent: undefined,
     };
   }
 
+  addContentToLocalStorage(content) {
+    if (!hasLocalStorage) {
```

Comment:
Why not adding locally in an array when local storage is not available? You can even namespace that differently in the select input